daemon/prompter.c

```diff
@@ -807,6 +809,52 @@ prompter_next_invocation(prompter_t *self)
 
 static void
 prompter_prompt_invocation_cb(GObject *obj, GAsyncResult *res, gpointer aptr)
+{
+    GCancellable    *cancellable = g_cancellable_new();
+    GDBusConnection *con  = G_DBUS_CONNECTION(obj);
+    prompter_t      *self = aptr;
+    GError          *err  = NULL;
+    GVariant        *rsp  = NULL;
+
+    if( !(rsp = g_dbus_connection_call_finish(con, res, &err)) ) {
+        if( err )
+            log_err("error reply: domain=%u code=%d message='%s'",
+                    (unsigned)err->domain, err->code, err->message);
+        else
+            log_err("null reply");
+        prompter_fail_invocation(self);
+    }
+    else if( !g_variant_check_format_string(rsp, "(o)", true) ) {
+        log_err("Invalid signature in reply: %s",
+                g_variant_get_type_string(rsp));
+        g_variant_unref(rsp);
+    } else {
+        gchar *object_path = NULL;
+        g_variant_get(rsp, "(o)", &object_path);
+        change_string(&self->prm_prompt, object_path);
+        change_cancellable(&self->prm_cancellable, cancellable);
+        g_dbus_connection_call(prompter_connection(self),
+                               WINDOWPROMPT_SERVICE,
+                               self->prm_prompt,
+                               WINDOWPROMPT_PROMPT_INTERFACE,
+                               WINDOWPROMPT_PROMPT_METHOD_WAIT,
+                               NULL,
+                               NULL,
+                               G_DBUS_CALL_FLAGS_NONE,
+                               G_MAXINT,
+                               cancellable,
```

Comment:
There is a hazard there that we would still need to `unref()` that cancellable or make that `change_cancellable()` to not to `ref()` it which would mean that all of the uses need changes. Anyway it seems that we have only two types of uses for `change_cancellable()`: either it's used to set a new cancellable that's not otherwise touched or to null an existing one. Maybe there is some chance for some cleanup there.

Reply:
Cleaned up cancellable use. There were fewer of them than I remembered.

Reply:
The function could be named change_cancellable_steal() to imply ref ownership transfer. A bit like how change_string() vs change_string_stral() already is.

Reply:
Changed. :ok_hand: